### ISSUE TITLE
Can´t see assets after Select/change default itens to show on Global View (memory) GLPI 9.5.1

### ISSUE THREAD

**Issue by monunes78** [glpi-project/glpi#7997]
After select assets memory to show on default view, no more itens show on default view.
`glpi_items_devicememories`

DEBUG
```
SQL Error "1064": You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'glpi_items_devicememories``.`size`) / COUNT(``glpi_items_devicememories``.`id...' at line 8 in query "SELECT DISTINCT `glpi_computers`.`id` AS id, 'admmarcio' AS currentuser, `glpi_computers`.`entities_id`, `glpi_computers`.`is_recursive`, `glpi_computers`.`name` AS `ITEM_Computer_1`, `glpi_computers`.`id` AS `ITEM_Computer_1_id`, `glpi_states`.`completename` AS `ITEM_Computer_31`, `glpi_manufacturers`.`name` AS `ITEM_Computer_23`, `glpi_computers`.`serial` AS `ITEM_Computer_5`, `glpi_computertypes`.`name` AS `ITEM_Computer_4`, `glpi_computermodels`.`name` AS `ITEM_Computer_40`, `glpi_operatingsystems_9719987b154aaf3b42c3db32aef59090`.`name` AS `ITEM_Computer_45`, `glpi_locations`.`completename` AS `ITEM_Computer_3`, `glpi_computers`.`date_mod` AS `ITEM_Computer_19`, GROUP_CONCAT(DISTINCT CONCAT(IFNULL(`glpi_deviceprocessors_7083fb7d2b7a8b8abd619678acc5b604`.`designation`, '__NULL__'), '$#$',`glpi_deviceprocessors_7083fb7d2b7a8b8abd619678acc5b604`.`id`) ORDER BY `glpi_deviceprocessors_7083fb7d2b7a8b8abd619678acc5b604`.`id` SEPARATOR '$$##$$') AS `ITEM_Computer_17`, (SUM(``glpi_items_devicememories``.`size`) / COUNT(``glpi_items_devicememories``.`id`)) * COUNT(DISTINCT ``glpi_items_devicememories``.`id`) AS `ITEM_Computer_111` FROM `glpi_computers`LEFT JOIN `glpi_states` ON (`glpi_computers`.`states_id` = `glpi_states`.`id` )LEFT JOIN `glpi_manufacturers` ON (`glpi_computers`.`manufacturers_id` = `glpi_manufacturers`.`id` )LEFT JOIN `glpi_computertypes` ON (`glpi_computers`.`computertypes_id` = `glpi_computertypes`.`id` )LEFT JOIN `glpi_computermodels` ON (`glpi_computers`.`computermodels_id` = `glpi_computermodels`.`id` ) LEFT JOIN `glpi_items_operatingsystems` ON (`glpi_computers`.`id` = `glpi_items_operatingsystems`.`items_id` AND `glpi_items_operatingsystems`.`itemtype` = 'Computer' ) LEFT JOIN `glpi_operatingsystems` AS `glpi_operatingsystems_9719987b154aaf3b42c3db32aef59090` ON (`glpi_items_operatingsystems`.`operatingsystems_id` = `glpi_operatingsystems_9719987b154aaf3b42c3db32aef59090`.`id` )LEFT JOIN `glpi_locations` ON (`glpi_computers`.`locations_id` = `glpi_locations`.`id` ) LEFT JOIN `glpi_items_deviceprocessors` ON (`glpi_computers`.`id` = `glpi_items_deviceprocessors`.`items_id` AND `glpi_items_deviceprocessors`.`itemtype` = 'Computer' ) LEFT JOIN `glpi_deviceprocessors` AS `glpi_deviceprocessors_7083fb7d2b7a8b8abd619678acc5b604` ON (`glpi_items_deviceprocessors`.`deviceprocessors_id` = `glpi_deviceprocessors_7083fb7d2b7a8b8abd619678acc5b604`.`id` ) LEFT JOIN `glpi_items_devicememories` ON (`glpi_computers`.`id` = `glpi_items_devicememories`.`items_id` AND `glpi_items_devicememories`.`itemtype` = 'Computer' ) WHERE `glpi_computers`.`is_deleted` = 0 AND `glpi_computers`.`is_template` = 0 GROUP BY `glpi_computers`.`id` ORDER BY ITEM_Computer_1 ASC LIMIT 0, 15"
```
I think this is the problem ... `glpi_items_devicememories` the two  ` 

Before
![image](https://user-images.githubusercontent.com/71788307/94057506-1536a780-fdb6-11ea-9db8-3b94c65e9bf7.png)
![image](https://user-images.githubusercontent.com/71788307/94057559-2b446800-fdb6-11ea-8e52-7464d920ae3e.png)

After
![image](https://user-images.githubusercontent.com/71788307/94056739-13b8af80-fdb5-11ea-97ba-2a385b0df3c7.png)

**Comment by trasher** [comment]
Please try with up to date current bugfixes branch (or [latest nightly](https://github.com/glpi-project/glpi/actions?query=workflow%3A%22GLPI+nightly+build%22)); as I've not been able to reproduce on my side.

**Comment by cedric-anne** [comment]
Hi,

Your problem seems to be the same as #7843, #7971 and #7870. We were not able to reproduce it on a freshly installed instance.

Looking at the code, it should not fail. We suspect a caching issue on PHP process.
Can you try to restart apache / php-fpm and see if problem persist ?

Regards

**Comment by monunes78** [comment]
> Please try with up to date current bugfixes branch (or [latest nightly](https://github.com/glpi-project/glpi/actions?query=workflow%3A%22GLPI+nightly+build%22)); as I've not been able to reproduce on my side.

Thanks for your help. A tryed the last build update but it doesen´t work.

**Comment by cedric-anne** [comment]
> > Please try with up to date current bugfixes branch (or [latest nightly](https://github.com/glpi-project/glpi/actions?query=workflow%3A%22GLPI+nightly+build%22)); as I've not been able to reproduce on my side.
> 
> Thanks for your help. A tryed the last build update but it doesen´t work.

Did you try to restart apache / php-fpm ?

**Comment by monunes78** [comment]
> > > Please try with up to date current bugfixes branch (or [latest nightly](https://github.com/glpi-project/glpi/actions?query=workflow%3A%22GLPI+nightly+build%22)); as I've not been able to reproduce on my side.
> > 
> > 
> > Thanks for your help. A tryed the last build update but it doesen´t work.
> 
> Did you try to restart apache / php-fpm ?

I tryed to restart the system, and aply this command "systemctl restart php-fpm", but not work for me.

**Comment by monunes78** [comment]
> > > > Please try with up to date current bugfixes branch (or [latest nightly](https://github.com/glpi-project/glpi/actions?query=workflow%3A%22GLPI+nightly+build%22)); as I've not been able to reproduce on my side.
> > > 
> > > 
> > > Thanks for your help. A tryed the last build update but it doesen´t work.
> > 
> > 
> > Did you try to restart apache / php-fpm ?
> 
> I tryed to restart the system, and aply this command "systemctl restart php-fpm", but not work for me.

I tryed this systemctl restart httpd,service.
How can I clean the php cache on centOS 7?

**Comment by monunes78** [comment]
I`am using fusion inventory for inventory assets.

**Comment by monunes78** [comment]
How can I reset this page visualization to default again?

**Comment by monunes78** [comment]
SOLVED
https://github.com/glpi-project/glpi/issues/7870#issuecomment-679227931

Just DOWNLOAD LAST GLPI VERSION AND replace the file search.class.php in the folder /glpi/inc

**Comment by cedric-anne** [comment]
Hi @monunes78 

In your issue title, you mentionned GLPI 9.5.1, then you said your bug was resolved by replacing the `search.class.php` by the one retrieved in the latest GLPI version. As GLPI 9.5.1 is the latest GLPI version, `search.class.php` should have same contents and its replacement should not change anything

As for the issue #7870, I do not understand why you had a buggy source code. Where did you get your GLPI source code which was containing the bug ? Can you attach the `search.class.php` you had in it ?
If there is some kind of unofficial GLPI package that contains bug, we would like to know it, in order to not waste time on a bug that cannot be reproduced on the official GLPI source code.

Regards

**Comment by monunes78** [comment]
Hi
Thank you for your help!

These are the files in the old version with the problem and the new one I used to solve the problem.

[new_search.class.txt](https://github.com/glpi-project/glpi/files/5282257/new_search.class.txt)
[old_search.class.txt](https://github.com/glpi-project/glpi/files/5282258/old_search.class.txt)

**Comment by cedric-anne** [comment]
OK, so I just take a look and it appears that you were using a `search.class.php` file from GLPI 9.4.3, patched by @stdonato.

When you have a bug on an application that you manually patched, please try to remove your patch before opening an issue. If bug comes from your patch, we waste time to try to understand something that cannot happens on the official package of the application.

`$ git diff 9.4.3 inc/search.class.php`
```diff
diff --git a/inc/search.class.php b/inc/search.class.php
old mode 100755
new mode 100644
index a81858ea3d..115ea0de98
--- a/inc/search.class.php
+++ b/inc/search.class.php
@@ -5686,19 +5686,30 @@ JAVASCRIPT;
                      $bar_color = 'yellow';
                   }
 
-                  if (!isset($so['datatype'])) {
-                     $so['datatype'] = 'progressbar';
-                  }
-
-                  $progressbar_data = [
-                     'text'         => Html::convDateTime($data[$ID][0]['name']),
-                     'percent'      => $percentage,
-                     'percent_text' => $percentage_text,
-                     'color'        => $bar_color
-                  ];
+                 // Stevenes Donato
+
+                  //bar colors
+                                               if($percentage == 100) { $cor = "progress-bar-danger"; }
+                                               if($percentage >= 80 and $percentage < 100) { $cor = "progress-bar-danger"; }
+                                               if($percentage > 51 and $percentage < 80) { $cor = "progress-bar-warning"; }
+                                               if($percentage > 0 and $percentage <= 50) { $cor = "progress-bar-success"; }
+                                               if($percentage < 0) { $cor = "progress-bar-success"; $percentage = 0; }
+
+
+                  //Calculate bar progress
+                  $out .= "<div class='center'>";
+                  $out .= "<div style='position:absolute; color:#000;'>&nbsp;".$percentage_text."%</div>";
+                  $out .= "<div class='progress' style='margin-top: 0px; margin-bottom: 0px; height: 14px;'>";
+                  $out .= "<div class='center progress-bar ".$cor." progress-bar-striped ' role='progressbar' aria-valuenow='".$percentage."' aria-valuemin='0' aria-valuemax='100'
+                   style='background-colorx: ".$color."; width: ".$percentage."%; height: 14px' > </div>";
+                  $out .= "</div>";
+                  $out .= "</div>";
+                  return $out;
                }
                break;
 
+               // Stevenes Donato
+
             case "glpi_softwarelicenses.number" :
                if ($data[$ID][0]['min'] == -1) {
                   return __('Unlimited');
```

**Comment by monunes78** [comment]
Sorry about that.
I installed the modifications and dashboard plugin thinking they are compatible.

(Modifications of login page, home and menu bar for GLPI 9.X)
https://github.com/stdonato/glpi-modifications/releases/tag/1.5.5
https://github.com/stdonato/glpi-dashboard/releases

I am really sorry for the loss of time, but I believe it can serve as support for other related problems in the future. When I removed the dashboards plugin the problem was really solved. I had previously tested just disable them.

**Comment by cedric-anne** [comment]
Hi,

glpi-modifications plugin replaces pieces of code of GLPI core during its installation. I was not aware of that, so, sorry if I were rude with you, as you were probably not aware that your GLPI files were "patched".

A new version of this plugin has been released for GLPI 9.5, but it seems that the scope of this plugin has been greatly reduced.

Regards

**Comment by monunes78** [comment]
Hi
No way were you rude to me.
Again I apologize for the inconvenience.
But I think it was the Dashboards plugin that caused the problem.
